Build RuneLite with Gradle

MarkKoester opened the pull request:
Using Gradle instead of Maven to build RuneLite reduces project build times and add more flexibilty for customizing the build process. It also offers a better user-experience than Maven since it has a more modern, discoverable CLI, and it ditches XML for its build files and uses a Groovy-based DSL instead.

A longer explanation of the benefits of using Gradle over Maven can be found in the [Gradle vs Maven comparison](https://gradle.org/maven-vs-gradle/) section of the Gradle docs.

## Results
Below is a comparison of build times using both Gradle and Maven. Results were obtained from a clean build and from an incremental build, where only one source file was changed.

Tests were skipped in all cases.

### Clean build
| Build system     | Time (minutes )|
|------------------|----------------|
| Maven            | 1:27           |
| Gradle           | 1:32           |

### Incremental build
| Build system     | Time (minutes )|
|------------------|----------------|
| Maven            | 1:03           |
| Gradle           | 0:18           |

Although build times are approximately the same for clean builds, incremental builds are significantly faster with Gradle. 🚀

## Change summary
- Remove `pom.xml` from all projects
- Add `build.gradle` to all projects
- Add package name to antlr grammer
- Remove cache test dependency on itself
- Remove `cache` project dependency from CacheClientTest.java
- Upgrade mapstruct to 1.3 (Needed to work with jdk8 and the current version is deprecated)
- Moved Checkstyle files to the `checkstyle` directory to work with gradle defaults
- Change path to suppressions.xml in `checkstyle.xml`
- Using palantir:git-version plugin to get git info (http-api)
- Changed some property names since Gradle interprets names separated by `.` to be accessing an object
- The `User-Agent` header string in the `http-api` library (`RuneLiteAPI.java`) was altered slightly since we can no longer use the Maven plugin for retrieving git information. It still includes all the same information as before (the commit hash and an indicator for whether the git tree is dirty), but it's formatted slightly differently. If the header needs to remain exactly the same it'll require a bit more effort, but if the header is just being logged for reproducing issues it should be fine.

jplsek committed:
inventory grid: add color config

MarkKoester committed:
Convert project to gradle

A catch for NullPointerException was added to OverlayIndex.java to fix
the build

MarkKoester committed:
Add copyright header to build files

f0rmatme commented:
hey man nice work, runeliteplus could use this. our client is less closed source so you can access our injector and make then necessary changes that you cannot do to runelite. 
https://github.com/runelite-extended/runelite/

also come hang out in our discord with other devs! 
https://discord.gg/jDCBh5J

Adam- committed:
client: accept custom javconfig url

Enriath committed:
item charges: add bracelet of slaughter and expeditious bracelet

Enriath committed:
slayer: remove bracelet charge tracking

Nightfirecat committed:
menu manager: add menu entries in insert order

Nightfirecat committed:
ProgressBar: Fix center label centering

BorderLayout does not guarantee equal sizing of its children, so having
a component centered between two components of unequal sizes will leave
it centered relative to their sizes, such as the following:

```
|-----BorderLayout-------------------------------------------|
||----WEST----||------CENTER------||----------EAST----------||
||------------||------------------||------------------------||
|------------------------------------------------------------|
```

Conversely, GridLayout uses fixed, equal-width columns for its children,
ensuring the center component will always occupy the center third of the
parent's dimensions.

P.S. This change causes labels to truncate earlier as they do not expand
beyond a third of the parent container's dimensions. Because the use
case for this component is with small labels, I do not anticipate this
will cause breakage.

Nightfirecat committed:
XpInfoBox: Give stats panel equal column widths

Nightfirecat committed:
item identification plugin: Add item mapping test

This will ensure any ImmutableMap building errors are caught at test
time rather than at runtime.

Enriath committed:
camera: add option to preserve pitch on compass look menu entries

Nightfirecat committed:
clues: Fix three step cryptic clue getLocations NPE

Prior to this commit, ThreeStepCrypticClue simply returned a
concatenation of the active step locations without filtering the mapped
locations in any way. This could lead to NPEs in the plugin as some
cryptic clues have null locations for steps which have no specific
location or have a variable location. This commit addresses this by
making CrypticClue's location field `@Nullable`, filtering null
locations from ThreeStepCrypticClue's getLocations stream, and by adding
a test to ensure ThreeStepCrypticClue's getLocations method cannot yield
any null entries in its return value.

Enriath committed:
world map overlay: make icons display fully on the map when edge snapped

Enriath committed:
world map mouse listener: consume click if icon is clicked on

This stops the map from being closed when clicking on an icon
overlapping the close button.

Broooklyn committed:
poh: Add icons for objects changed with Goblin Decorations

Broooklyn committed:
worldmap: Add TzHaar-Ket-Rak's Challenges minigame location

JoRouss committed:
party: add party panel

LlemonDuck committed:
Add Below Ice Mountain quest

Add BIM to the world map plugin quest start locations, and the Camdozaal
dungeon to the skybox and discord regions

Co-authored-by: Adam <Adam@sigterm.info>

FlashBasedGod committed:
ground items: Add Tempoross instance to normal despawn timers

aHooder committed:
gpu: workaround for forced anti-aliasing

Forced anti-aliasing by the GPU applies multisampling to the default
framebuffer, and when blitting between two multisampled FBOs, the number
of samples in both FBOs has to match.

In order to make sure our anti-aliasing FBO has the same number of
samples as the default FBO, we bind the default FBO and check whether
its number of samples is non-zero. If it is, use this as the number of
samples for our anti-aliasing FBO instead of getting it from the
configured anti-aliasing mode.

Co-authored-by: Adam <Adam@sigterm.info>

Adam- committed:
friends chat: use friendschatinfo color for join/part messages

Adam- committed:
chat message manager: add default color for friendschatnotification chat type

This allows the config value for friends chat notifications to just
default to null, allowing plugins to detect whether or not we want to
override the default color for friends chat notification messages.

amitgoldGH committed:
SkillChallengeClue: Support inactive crystal tools (#13517)

Nightfirecat committed:
HotColdLocation: Fix Seers' Village bank spot

Ref: #9601

Nick-Anzalone committed:
skill calc: Add battlesaff to fletching calculator (#13525)

Commit:
examine: Improve plugin description and tags (#13399)

Nightfirecat committed:
status bars: Add counter text drop shadow

This drop shadow was present prior to the plugin refactor in
1165780f427e1d9f86f2d658b050bf04a7ae237e which erroneously removed it.

Enriath committed:
Update discord invite links to start users in #welcome (#13549)

Cyborger1 committed:
SkillChallengeClue: Add Gold shade keys to Fiyr shade step (#13260)

LlemonDuck committed:
clues: Fix location of Captain Bleemadge (#13531)

Adam- committed:
agility plugin: add rellekka lighthouse obstacles

Adam- committed:
info: simplify language surrounding config import

JamesShelton140 committed:
xptracker: add option to reset per hour rates

JamesShelton140 committed:
xptracker: add auto reset per hour rates after set time

emiljensen2 committed:
loot tracker: support tob lobby reward chest

Adam- committed:
item charges: simplify infobox creation logic

This looks at the worn items and matches them to known items with
charges, instead of checking the worn items for each known item with
charges. Additionally it associates guthix rest to potions, because
previously they did not have a config option, and adds an infobox for
the chronicle.

RuneLogApp committed:
music plugin: add null check to SettingsSideSlider icon on shutdown

Adam- committed:
clientloader: don't use fallback config or client when jav_config is manually specified

Commit:
Release 1.7.7

Commit:
Bump for 1.7.8-SNAPSHOT

Adam- committed:
item charges: add toggle for guthix rest

Since guthix rest are not visibly different, but potions are, there are requests to restore the old behavior of having guthix rest on but potions off

Adam- committed:
item charges: split up test into separate methods

Adam- committed:
chat message manager: fix recoloring own name with spaces

Adam- committed:
xptracker: use enum map for actions and other cleanup

Adam- committed:
party: only send location update when the location changes

Adam- committed:
info panel: adjust nav button priority

Adam- committed:
ui: queue repaint when progressbar is edited

Enriath committed:
clues: support imcando hammer in skill challenges

Nightfirecat committed:
Merge pull request #13609 from Hydrox6/clues-imcando-hammer

clues: support imcando hammer in skill challenges

Adam- committed:
discord plugin: sanity check user id and avatar id before building url

Adam- committed:
party service: sanitize user names

Commit:
Update NPC IDs to 2021-5-19

Commit:
Update Item variations to 2021-5-19

Commit:
Update Item IDs to 2021-5-19

Commit:
Update Object IDs to 2021-5-19

Commit:
Update Widget IDs to 2021-5-19

Commit:
Update Scripts to 2021-5-19

Updated script does not assemble FriendsChatSendKick.rs2asm

abextm committed:
friendschat: remove Tab Chat option

this is now included in the vanilla client

abextm committed:
friendschat: remove user count

this widget is gone, we can find a new place for it in the future maybe

emerald000 committed:
Rename Armadylian Guards to Armadylean Guards

Commit:
Update 196

abextm committed:
gpu: increase texture limit

abextm committed:
friendschat: remove unused title strings

Commit:
Release 1.7.8

Commit:
Bump for 1.7.9-SNAPSHOT

abextm committed:
SpriteID: update to 2021-5-19

abextm committed:
screenshot: use new report button offset

abextm committed:
interfacestyles: update chatbox sprite ids

These are all the wrong size, but they aren't horribly broken like this

abextm committed:
rl-api: import getTopLevelInterfaceId

abextm committed:
screenshot: don't attempt to draw a date when there is no report button

previously this would just silently ignore the screenshot if you were
on the orb of oculus or fullscreen world map interfaces until you were
back onto one of the normal tlis where the overlay is rendered.

Adam- committed:
chat timestamps: add timestamps for all message types

Adam- committed:
minimap: add clan chat dot color configuration

JoRouss committed:
party: readd overlay leave party menu entry

abextm committed:
scripts/SkillTabBuilder: fix unbalanced stack

Adam- committed:
twitch: change chat trigger to /t

// was taken by clan chat with the recent update

abextm committed:
cache: update cs2 opcodes

abextm committed:
rl-client: update cs2 overlay opcodes

abextm committed:
rl-client: cleanup cs2 overlays

Adam- committed:
chat history: support clan chat

Additionally fix the TAB_PRIVATE widget mapping which is incorrect with the
latest update. The old friends chat tab has also gained a Clear history
option so we no longer need to add it.

Adam- committed:
chat history: rebuild pmbox when all is cleared

Additionally it is unnecessary to manually remove messages when the clear option is vanilla, since it will do it already

Adam- committed:
friendschat: readd user count

Also readd the Recent FCs title

Adam- committed:
api: add getOnVarTransmitListener to Widget

Enriath committed:
clues: count bird glove as being correct for the weapon requirement

Adam- committed:
api: add enum size

Adam- committed:
api: correct clans chattypes

Adam- committed:
devtools: add modicons command

Adam- committed:
friendchatmanager: rename to chaticonmanager

FriendsChatManager predates the api having an ability to look up clan
members by name, without doing string matching, hence the caching. This
is no longer necessary as we can do efficient lookups of players by name
now.

Also remove the old isMember utility function which is just duplicated
from the friends chat manager.

Adam- committed:
Change config name of party plugin to disable by default

Since the plugin was rewritten and changed to disable by default, it will not retroactively apply to existing configs without this

abextm committed:
ChatboxPanelManager: use ScriptPreFired

abextm committed:
timestamp: use less callbacks in script overlays

all of these separate callbacks are error prone to update and hard to
test

Commit:
Update Object IDs to 2021-05-26-rev196

Commit:
Update NPC IDs to 2021-05-26-rev196

Commit:
Update Scripts to 2021-05-26-rev196

abextm committed:
Merge pull request #13637 from abextm/cache-code-2021-05-26-rev196

Cache code 2021 05 26 rev196

Commit:
Release 1.7.9

Commit:
Bump for 1.7.10-SNAPSHOT

Enriath committed:
skybox: add clan hall

Adam- committed:
api: add clan api

Adam- committed:
chat colors: add clan recolors

Adam- committed:
world hopper: add hop-to clan member

Adam- committed:
player indicators: add clans

Adam- committed:
chat filter: add clan chat

Adam- committed:
emoji plugin: add clan chat

Adam- committed:
chat notifications: add clan chat

Adam- committed:
chat commands: add clan chat

Adam- committed:
chat icon manager: add clan chat icons

Adam- committed:
Allow rs.version to be a double

This allows sub-versions of the same RS revision

LlemonDuck committed:
Reset spec counter when leaving instanced fights

When leaving Vorkath or Alchemical Hydra fights, the spec counter does not reset, despite the boss resetting their stats. This could cause the spec counter to add the player's next trip spec to that, giving an inaccurate total spec count. This clears that counter when leaving the vork + hydra region instances.

Adam- committed:
runelite-api: add clan member join and leave events

Adam- committed:
friends chat: rename to chatchannel

This makes the friends chat plugin applicable to all chat channels, and adds join/leave notifications for clan chats

Adam- committed:
chat channel: add typing modes

This allows changing the target of typed messages via /f, /c, /g, and
/p. All further messages go to the given target until a new target is
specified.

Nightfirecat committed:
Merge pull request #13640 from Hydrox6/skybox-clan

skybox: add clan hall

Commit:
Update 196.2

Commit:
Release 1.7.10

Commit:
Bump for 1.7.11-SNAPSHOT

Adam- committed:
Strip tags when doing friends chat lookups

The previous friend chat manager would do this when loading a name into the cache, but now that the cache is gone, we need to do it prior to the lookup

Adam- committed:
team: add clan member counter

Adam- committed:
team capes: rename to team

Move friends list indicator from chatchannel to team plugin

Adam- committed:
api: add clan channel changed event

abextm committed:
rl-client: never cache 4/5xx requests

Adam- committed:
chatchannel: fix clan join/leave timeouts with fc join/leave off

Adam- committed:
chat filter: fix collapsing replayed public chat

Public chat is both replayed and also collapsed. Since replayed messages
don't send chat message events, all replayed messages would be filtered
due to the last message id not being updated for any of them. Instead,
set all stored messages ids to -1 on login, since the ids are
invalidated anyway, and then use it to not collapse replayed chat.

Adam- committed:
sprite manager: use constructor injection

Also make cache private

Adam- committed:
Use junit rule for mock webserver

Adam- committed:
camera: add zoom slider tooltip when dragging

Co-authored-by: superiorser9 <superiorser9@gmail.com>

Adam- committed:
chat channel: add target mode config

LlemonDuck committed:
camera: prevent pitch increment with "preserve pitch" enabled

When clicking the compass, the vanilla camera will jitter randomly
within 5 values. With "compassLookPreservePitch" enabled, spam clicking
the compass would cause your camera pitch to rise repeatedly by that
random jitter. This places the camera pitch restore callback after the
random jitter, to eliminate that discrepancy.

Commit:
Update Scripts to 2021-6-3

Commit:
Update Item variations to 2021-06-03-rev196

Commit:
Update Item IDs to 2021-06-03-rev196

Commit:
Update NPC IDs to 2021-06-03-rev196

Commit:
Update Object IDs to 2021-06-03-rev196

Commit:
Release 1.7.11

Commit:
Bump for 1.7.12-SNAPSHOT

Adam- committed:
http-api: add tob hm hiscores

Adam- committed:
hiscore panel: add tob hard mode

Adam- committed:
chat filter: remove filter login

This has been added as a setting ingame with the latest update

Adam- committed:
mta: clear hint arrow when leaving enchantment and alchemist room

LlemonDuck committed:
idle notifier: reset six hour notif on startup

Six hour notification could trigger incorrectly when the plugin was enabled if the player disabled -> logged out -> logged in -> enabled, since the notification timer was not reset.

Adam- committed:
chat commands: fix tob hm/sm kc/pb tracking

The : in the boss name breaks for logged in users, so remap it to a
clean name instead. Also add short names.

Adam- committed:
config manager: disallow : in key names

The config service rewrites . -> : and : -> . for storage since mongodb
does not support . in field names. This causes any keys set with : to
get changed after going to the config service and back.

SkylerPIlot committed:
spec counter: add Alchemical Hydra

This keeps the counter up when the hydra changes phases

Signed-off-by: SkylerPIlot <SkylerParthemore@gmail.com>

ThePharros committed:
screenshot: Add option to screenshot collection log entries (#13625)

Adam- committed:
chat colors: fix clan chat message highlight keys

The old friends chat uses "clan chat" as its key name for historical
reasons, and so the new clan chat uses "clan"

Nightfirecat committed:
HotColdLocation: Add clue type field

Adam- committed:
api: add setCameraYawTarget

Adam- committed:
camera: add option to preserve yaw when world hopping

Adam- committed:
chat commands: update tob new pb pattern

The old text (Personal best!) is no longer used

Adam- committed:
gpu: use renderbuffer for aa fbo

Previously this bound first a render buffer and then a texture to the aa fbo color attachment0, making the renderbuffer unused. Just remove the texture, causing the render buffer to be used.

Adam- committed:
menu swapper: add con cape tele to poh

Cyborger1 committed:
ItemIdentification: Add Ghorrock tablet

Nightfirecat committed:
Merge pull request #13718 from Cyborger1/ghorrock-tab-identification

ItemIdentification: Add Ghorrock tablet

Cyborger1 committed:
ChatHistory: Fix "Copy to Clipboard" after clan update

abextm committed:
cache: add 197 opcodes

Adam- committed:
chat commands: update tob duration pattern

Nightfirecat committed:
Merge pull request #9021 from jplsek/inv-grid-color-config

Inventory Grid Plugin: Add color configuration

Nightfirecat committed:
Merge pull request #13499 from FlashBasedGod/tempoross-timer-red

Tempoross timer bug

Nightfirecat committed:
Merge pull request #13680 from LlemonDuck/fix-camera-pitch-jitter

camera: prevent pitch increment with "preserve pitch" enabled

abextm committed:
client: correctly use Runnables::doNothing

passing it as a method reference complete defeats the point of the
method as it desugars to:
option("foo", () ->
{
  Runnable unused = Runnables.doNothing();
});

Adam- committed:
ground markers: add configurable border width

Adam- committed:
api: add widget line height accessors

tanlines committed:
itemstats: Fix stamina restore with Ring of Endurance (#13692)

Cyborger1 committed:
hiscore: add lookup clan member (#13706)

Cyborger1 committed:
clues: Update description for coordinate south-west of Ferox

Adam- committed:
menu swapper: add achievement diary items with non-standard teleport options

Co-authored-by: GG46KG <cvernino@att.net>

Adam- committed:
Remove examine api

The non-items examine info has been broken for years, and the wiki already has item examines anyway, so we can use those in the future if we need them.

Adam- committed:
xp drops: add option to override standard client xp drop colors

Co-authored-by: Zeusay <kentgamesdesign@Gmail.com>

Adam- committed:
screenshot plugin: consolidate kill count fields

Adam- committed:
screenshot plugin: make subdir strings constants

Adam- committed:
screenshot plugin: add tob story and hard mode screenshots

Adam- committed:
screenshot plugin: name tob hm/sm files differently

steven10172 committed:
loot tracker: add Wintertodt supply crate

emiljensen2 committed:
ItemMapping: add new tob kits

Add holy/sanguine scythe, sang, and rapier varients including charged/uncharged versions.

Nightfirecat committed:
clues: Update and add clues for 6/16 Shayzien rework

Nightfirecat committed:
Merge pull request #13735 from emiljensen2/patch-3

Nightfirecat committed:
Merge pull request #13727 from Cyborger1/ferox-coordinate-hard-clue

clues: Update description for coordinate south-west of Ferox

Bird-Hands committed:
npc indicators: add configurable border width

Nightfirecat committed:
object indicators: add configurable border width

Nightfirecat committed:
Merge pull request #13712 from Bird-Hands/NPCIndicators

npc indicators: add outline width

danshea00 committed:
screenshot: Use separate directory for tournament/beta worlds (#13631)

Nightfirecat committed:
Merge pull request #13629 from Hydrox6/clues-bird-is-empty

clues: count bird glove as being correct for the weapon requirement

Commit:
Update Item variations to 2021-6-16

Commit:
Update Object IDs to 2021-6-16

Commit:
Update Item IDs to 2021-6-16

Commit:
Update NPC IDs to 2021-6-16

Commit:
Update Widget IDs to 2021-6-16

abextm committed:
rl-api: update Quest enum to 2021-6-16

Adam- committed:
loot tracker: add reset warning to event reset option

Commit:
Update 197

Commit:
Release 1.7.12

Commit:
Bump for 1.7.13-SNAPSHOT

Enriath committed:
loginscreen: add a kingdom divided login screen

abextm committed:
worldmap: add shayzien rework locations

Enriath committed:
api: correct imcando hammer construction animation

abextm committed:
agility: add shayzien course

abextm committed:
barbarianassault: remove dead overlay menuops

menu options are never added on dynamic widgets since they don't have
bounding boxes

abextm committed:
runelite-client: use menu entries for focusing world map points

The previous way had usability issues since it would cover up the close
button, consumed clicks for no reason on any icon, making it difficult
to pan around the map when zoomed out, and had no tooltips.

It previously also accessed widgets on the EDT, which is a threading
issue.

petter committed:
clues: Add Giant's Den clue

Nightfirecat committed:
Merge pull request #13744 from petter/giants-den-cryptic-clue

clues: Add the new cryptic clue at Giant's Den

Nightfirecat committed:
clues: Add new A Kingdom Divided update clues

Adam- committed:
discord: fix npe from menu event timing out with elapsed total time

Setting the start time to null isn't necessary, since IN_MENU is marked
shouldRestart, it always resets the start time when the status is changed to
menu. This was clearing it, forcing a presence update, and then erroring when
trying to compute the original start time. It also would have never really unset,
even if it wasn't erroring later, due to IN_MENU also being unclearable, and
so would have never been removed in the first place.

Additionally the status reset test is also wrong since it is testing the end
timestamp which has never been used by us, making the test useless.

Nightfirecat committed:
clues: Fix clues changed with A Kingdom Divided update

Nightfirecat committed:
Merge pull request #13743 from abextm/akd

Enriath committed:
timers: add duration and cooldown timers for arceuus spells

Co-authored-by: Adam <Adam@sigterm.info>

Enriath committed:
loot tracker: check for correct menu option type when deciding event

Adam- committed:
chat icon manager: update clan rank to idx mapping

Nobody was added with key -6

Enriath committed:
skill calc: add demonic ashes

Enriath committed:
skill calc: add shayzien courses

Enriath committed:
skill calc: add new arceuus spells

Adam- committed:
menu swapper: add enter-crypt to quick swap

This is for Sarachnis

Nightfirecat committed:
Merge pull request #13741 from Hydrox6/login-screen-new

loginscreen: add a kingdom divided login screen

Nightfirecat committed:
agility: Fix Shayzien basic course total xp

The final obstacle of the Shayzien basic course grants 92.2xp, not a
flat 92.

Nightfirecat committed:
Merge pull request #13762 from Hydrox6/skill-calc-akd

Adam- committed:
api: add actor overhead cycle accessors

Enriath committed:
api: fix hitsplat sprite ids

Nightfirecat committed:
Merge pull request #13768 from Hydrox6/venom-hitsplat

andmcadams committed:
crowdsourcing/dialog: track start & end of conversations (#13763)

Commit:
Update Item IDs to 2021-6-23

Commit:
Update Object IDs to 2021-6-23

Commit:
Update Widget IDs to 2021-6-23

Commit:
Update Scripts to 2021-6-23

SirWrain committed:
achievementdiary: Fix hard Lumbridge diary entry text (#13689)

Nightfirecat committed:
EmoteClue: Fix Shayzien war tent clue text

abextm committed:
worldmap: include names for teleport icons

Commit:
Release 1.7.13

Commit:
Bump for 1.7.14-SNAPSHOT

Enriath committed:
widgetoverlay: add back removed WidgetOverlays

These were removed in commit d566998179d3985242a451e484467e6023012374
due to major interface changes from Jagex

Enriath committed:
combatlevel: fix attack range indicator

Co-authored-by: Jordan Atwood <jordan.atwood423@gmail.com>

Enriath committed:
clues: fix hot/cold enemies for updated kourend locations

The shayzien overhaul caused the area to become single-way.
The area around the mess hall was also made single-way for some reason.

Nightfirecat committed:
Merge pull request #13784 from Hydrox6/clues-shayzien

clues: fix hot/cold enemies for updated kourend locations

Enriath committed:
timers: make thrall timer use boosted magic level

turns out that thrall spells are seemingly the only arceuus spell with a
magic-level based duration that uses the boosted level, at least out of
the ones we have timers for

Nightfirecat committed:
Merge pull request #13776 from Hydrox6/widgetoverlay-readd

WooxSolo committed:
api: add actor current orientation

Co-authored-by: Adam <Adam@sigterm.info>

WooxSolo committed:
api: add decorative object offsets

Co-authored-by: Adam <Adam@sigterm.info>

WooxSolo committed:
api: add game object model orientation

Co-authored-by: Adam <Adam@sigterm.info>

geheur committed:
mta: update alchemy room widgets ids

WooxSolo committed:
Add model outline renderer

Add the option to outline models to the npc and object indicator plugins

Co-authored-by: Adam <Adam@sigterm.info>
Co-authored-by: Max Weber <mii7303@gmail.com>

Commit:
Update Item IDs to 2021-6-30

Commit:
Update Object IDs to 2021-6-30

Commit:
Update NPC IDs to 2021-6-30

Commit:
Update Item variations to 2021-6-30

Commit:
Update Scripts to 2021-6-30

Commit:
Update Widget IDs to 2021-6-30

abextm committed:
Update script arguments to 2021-6-30

Adam- committed:
tile indicators: add border width config

abextm committed:
http-api, rl-client: add phosani's nightmare to hiscores

Commit:
Release 1.7.14

Commit:
Bump for 1.7.15-SNAPSHOT

abextm committed:
http-service: fix hiscore test

MarkKoester committed:
Update gradle to 7.1
